### PR TITLE
Do not select the default key with --ssh=agent

### PR DESCRIPTION
**Pull request by olegmlsn**
should address #170 

If no ssh key is specified and the `--ssh-agent` option is specified, do not select the default key

### CODE REVIEW

File: cmd/spot/main_test.go [hunk 1]
```diff
@@ -441,7 +441,7 @@
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			key, err := sshKey(tc.opts.SSHKey, &tc.conf)
+			key, err := sshKey(tc.opts.SSHAgent, tc.opts.SSHKey, &tc.conf)
```
Comment (umputun) [inline cmd/spot/main_test.go:444]: would be nice to add another test case above with opts.ssh-agent set

**Review by umputun — COMMENTED** [review]
thx. a small thing what is missing - a small test case for the change

**Comment by olegmlsn** [comment]
Evil linter ). I couldn't beat him. Nolint?
```go
//nolint:revive //Evil linter
```

**Comment by umputun** [comment]
re: linter thing - I have updated master, pls rebase your branch and it should pass

**Comment by umputun** [comment]
LGTM, thx

**Pull request merged**